sdk/core/ts-http-runtime/scripts/azure-diff.ts

```diff
@@ -0,0 +1,182 @@
+ *
+ * For each source file in the unbranded package, we will loop through these entries and find the first matching key. A key is considered
+ * a match if the unbranded file's path starts with the key.
+ *
+ * For example, any file under src/logger will map to ../logger/src (the Azure logger package), except for src/logger/logger.ts, which will get mapped
+ * to ../logger/src/index.ts, since that entry comes before src/logger in the object.
+ */
+const UNBRANDED_TO_AZURE_MAPPINGS = {
+  // Logger index file was renamed
+  "src/logger/logger.ts": "../logger/src/index.ts",
+
+  "src/logger": "../logger/src",
+  "src/auth": "../core-auth/src",
+  "src/abort-controller": "../abort-controller/src",
+  "src/client": "../core-client-rest/src",
+  "src/policies": "../core-rest-pipeline/src/policies",
+  "src/retryStrategies": "../core-rest-pipeline/src/retryStrategies",
+  "src/tracing": "../core-tracing/src",
+  "src/util": "../core-util/src",
+
+  src: "../core-rest-pipeline/src",
+};
+
+const {
+  values: { update = false, "output-path": outputPath = "./review/azure-core-comparison.diff" },
+} = parseArgs({
+  options: {
+    update: {
+      type: "boolean",
+    },
+    "output-path": {
+      type: "string",
+      short: "o",
+    },
+  },
+});
+
+/**
+ * Recursively find all files in the directory
+ * @param dir - Directory to find files in
+ * @returns - an array containing paths to all files in the directory
+ */
+async function find(dir: string): Promise<string[]> {
+  const files = await fs.readdir(dir);
+  const allFiles: string[] = [];
+  for (const filename of files) {
+    const file = path.join(dir, filename);
+    const stat = await fs.stat(file);
+    if (stat.isDirectory()) {
+      allFiles.push(...(await find(file)));
+    } else {
+      allFiles.push(file);
+    }
+  }
+
+  return allFiles;
+}
+
+/**
+ * Returns a list of all unbranded source files and their azure counterparts if any
+ * @returns - list of tuples [unbranded file, azure counterpart]. The azure counterpart may be undefined if no counterpart can be found
+ */
+async function getFileMappings(): Promise<[string, string | undefined][]> {
+  const result: [string, string | undefined][] = [];
+  const files = await find("src");
+
+  for (const unbrandedLocation of files) {
+    const pathMatches = Object.keys(UNBRANDED_TO_AZURE_MAPPINGS).filter((x) =>
+      unbrandedLocation.startsWith(x),
+    );
+
+    let found = false;
+    for (const pathMatch of pathMatches) {
+      const azureLocation =
+        UNBRANDED_TO_AZURE_MAPPINGS[pathMatch] + unbrandedLocation.substring(pathMatch.length);
+
+      let exists: boolean;
+      try {
+        await fs.access(azureLocation);
+        exists = true;
+      } catch {
+        exists = false;
+      }
+
+      if (exists) {
+        result.push([unbrandedLocation, azureLocation]);
+        found = true;
+        break;
+      }
+    }
+
+    if (!found) {
```

Comment:
do you think it would be useful to track the list of files that in azure core but not in unbranded? We could probably infer added files via related changes though.